File: hash_table/2085_count_common_words_with_one_occurrence/2085_count_common_words_with_one_occurrence.py
class Solution:
    def countWords(self, words1: List[str], words2: List[str]) -> int:
        freq1 = Counter(words1)   # words1 中字符串的出现次数
        freq2 = Counter(words2)   # words2 中字符串的出现次数
        res = 0   # 出现过一次的公共字符串个数
        for word1 in freq1.keys():
            # 遍历 words1 出现的字符并判断是否满足要求
            if freq1[word1] == 1 and freq2[word1] == 1:
                res += 1
        return res

File: hash_table/2085_count_common_words_with_one_occurrence/2085_count_common_words_with_one_occurrence.go
package main

func countWords(words1 []string, words2 []string) int {
	result := 0
	cnt1 := make(map[string]int)
	cnt2 := make(map[string]int)
	for _, w := range words1 {
		cnt1[w] += 1
	}
	for _, w := range words2 {
		cnt2[w] += 1
	}
	for k, v := range cnt1 {
		if v == 1 && cnt2[k] == 1 {
			result += 1
		}
	}
	return result

}
